src/DataTable/tableReducer.js

```diff
@@ -22,6 +22,7 @@ export function tableReducer(state, action) {
         sortColumn,
         selectedColumn,
         sortDirection,
+        currentPage: 1,
```

Comment:
Can open this as a separate PR as it is a different "feature"

Reply:
I did, here: https://github.com/jbetancur/react-data-table-component/pull/172

I presumed it would be merged by the time this one was, so I didn't bother extracting it from this one.

Reply:
This is a bigger change and I want to get your sorting fix into a 3.0.3 release :)

I'll need some time to review this PR today or tomorrow

Reply:
No problem, the sorting fix was really what I needed.  The search box feature I already have production ready on my end implemented on top of the library, but since I'd already done it and it's an obvious feature I'm sure putting it in the main library will be useful for someone.

Reply:
I thought about this a bit and instead of changing the source, I prefer the approach you took to add a custom search in the subHeader.  There are just so many ways someone would want to customize this that I think the subHeader `hook`  is the place to go. Additionally, we would need to handle the text input where people would like to use a 3rd party UI control; which only adds complexity to this library.

**Sidenote** and related to this, I've also considered column filtered fields where one could have an input on each column, but this suffers from the complexity issue above. It might be interesting though to allow people to place whatever component they want similar to `subHeaderComponent` for that scenario. and a column.filter: true.